src/Tester.ts

```diff
@@ -10,75 +10,410 @@ After the test a test report can be generated and analyzed to get more meaning o
+                    container.subscriptionReport.passed = true
+                    container.subscriptionReport.result = new Result(200)
+                    await Promise.race([Utils.sleepInMs(eventConfig.MsListen), earlyListenTimeout])
+                    break
+            }
+            return
+        }
+
+        /**
+         * Records received data and checks if it is valid.
+         * @param receivedData The received data package.
+         */
+        async function handleReceivedData(receivedData: any): Promise<void> {
+            const receivedTimeStamp = new Date()
+            ++indexOfEventData
+            // Stop recording if maximum number of recorded data packages is reached.
+            if (eventConfig.MaxAmountRecvData != null) {
+                if (indexOfEventData >= eventConfig.MaxAmountRecvData) {
+                    // Stop Listening for Data.
+                    earlyListenTimeout.resolve()
+                    return
+                }
+            }
+            // Stop handling if sent TD does not have "data" for event.
+            if (testMode == Utils.InteractionType.Event && !interaction.hasOwnProperty("data")) {
+                // receivedData === undefined if event was emitted without payload. This is correct as long as event has no "data" property.
+                if (receivedData === undefined) {
+                    self.log("Received as expected empty event with no payload.")
+                    const result = new Result(200, "Received as expected empty event with no payload.")
+                    container.eventDataReport.received.push(new EventData(receivedTimeStamp, receivedData, result))
+                    return
+                }
+                // Received unexpected event data (no "data" property for this event in the TD).
+                container.passed = false
+                container.eventDataReport.passed = false
+                try {
+                    const temp: JSON = receivedData
+                } catch (error) {
+                    // received data is not JSON
+                    self.log(
+                        receivedDataMsg +
+                            "[index: " +
+                            indexOfEventData +
+                            "]: received unexpected (no data property for this even in the TD), non JSON conformal data."
+                    )
+                    const result = new Result(98, "Received unexpected (no data property for this event in TD), non JSON conformal return value.")
+                    container.eventDataReport.received.push(
+                        new EventData(
+                            receivedTimeStamp,
+                            JSON.parse(JSON.stringify("NOT the actual data: The received data was not JSON conformal: " + error)),
+                            result
+                        )
+                    )
+                    return
```

Comment:
@egekorkan this refers to the question where you wrote you did not unerstand it. I need feedback concerning this section (from `if (testMode == Utils.InteractionType.Event && !interaction.hasOwnProperty("data")) {` up to here. I added an extra JSON check to make sure I do send a non JSON value...

I will ask pretty much the same question on affected part of testAction().

Reply:
Ah ok, all good!